haskell/src/Main.hs

```diff
@@ -0,0 +1,129 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ExtendedDefaultRules #-}
+
+import qualified Database.MongoDB as Mongo
+import qualified Data.Aeson as A
+import Network.HTTP.Server
+import Network.HTTP.Server.Logger
+import Network.URL as URL hiding (host)
+import Codec.Binary.UTF8.String
+import Data.AesonBson (aesonify)
+import Data.List (isPrefixOf)
+import Data.Maybe (fromMaybe)
+
+-- import Network.HTTP.Server.HtmlForm as Form
+-- import Text.JSON
+-- import Text.JSON.String (runGetJSON)
+-- import Text.XHtml hiding (URL, select)
+-- import Control.Exception (try, SomeException)
+-- import System.FilePath
+{-- MAIN ---------------------------------------------------------------------}
+
+type ServerConfig = Config
+data AppConfig = AppConfig { db :: Mongo.Database
+                           , coll :: Mongo.Collection
+                           , address :: String
+                           , token :: String }
+
+serverConfig :: ServerConfig
+serverConfig = defaultConfig { srvLog = stdLogger, srvPort = 8080 }
+
+appConfig :: AppConfig
+appConfig = AppConfig { db = "slackbot"
+                      , coll = "testData"
+                      , address = "127.0.0.1"
+                      , token = "m3ABP8zIiY5wzJAtuJcwffji" }
```

Comment:
What do you think if we have this read from external source i.e env var or file, etc. rather then hard-coded? As this might be a security issue